src/transport.rs

```diff
@@ -2,6 +2,15 @@ use std::io;
 use std::io::{Read, Write};
 use std::net;
 use std::net::TcpStream;
+#[cfg(feature = "ssl")]
+use openssl::ssl::{SslStream, SslConnector};
+
+pub  trait  CDRSTransport: Sized+Read+Write+Send+Sync{
+
+    fn try_clone(&self) -> io::Result<Self>;
+    fn close(&mut self, close: net::Shutdown) -> io::Result<()>;
+}
+
 
 #[derive(Debug)]
 pub struct Transport {
```

Comment:
Could you also rename it to `TransportTcp` so we to have `TransportTcp` and `TransportTls`. That would be consistent from naming point of view.

Also please make sure that Readme reflects these changes.